packages/ui/src/components/Visualization/Custom/CustomNode.tsx

```diff
@@ -15,10 +15,11 @@ interface CustomNodeProps extends WithSelectionProps {
 
 const CustomNode: FunctionComponent<CustomNodeProps> = ({ element, ...rest }) => {
   const vizNode = element.getData()?.vizNode;
+  const label = vizNode?.getNodeLabel();
 
   return (
-    <DefaultNode element={element} showStatusDecorator {...rest}>
-      <g data-testid={`custom-node__${vizNode?.id}`}>
+    <DefaultNode element={element} label={label} truncateLength={15} {...rest} showStatusDecorator>
+      <g data-testid={`custom-node__${vizNode?.id}`} data-nodelabel={label}>
```

Comment:
@tplevko Since long labels might be truncated, the entire label will be held in the `data-nodelabel` attribute, I updated the e2e commands to reflect this.